File: pset6/credit/credit.py
from cs50 import get_int

def main():
    number = get_int("Number: ")

    number_1 = number
    number_2 = int(number / 10)

    firstTwo = number
    firstOne = 0

    sum_1 = 0
    sum_2 = 0

    length = len(str(number))

    #Getting firstTwo & firstOne
    while firstTwo >= 100:
        firstTwo = int(firstTwo / 10)
    firstOne = int(firstTwo / 10)

    #From last digit
    while number_1 > 0:
        lastDigit = number_1 % 10
        sum_1 += lastDigit
        number_1 = int(number_1 / 100)

    #From second-last digit
    while number_2 > 0:
        lastDigit = number_2 % 10
        sum_2 += magic(lastDigit)
        number_2 = int(number_2 / 100)

    total = sum_1 + sum_2
    if total % 10 == 0:

        if length >= 13 and length <= 16:

            if firstTwo == 34 or firstTwo == 37:
                print("AMEX")

            elif firstTwo >= 51 and firstTwo <= 55:
                print("MASTERCARD")

            elif firstOne == 4:
                print("VISA")

            else:
                print("INVALID")

        else:
            print("INVALID")

    else:
        print("INVALID")


def magic(lastDigit):
    lastDigit *= 2
    if lastDigit >= 10:
        first = int(lastDigit / 10)
        second = lastDigit % 10
        result = first + second
        return result
    else:
        return lastDigit

if __name__ == "__main__":
    main()

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long number = get_long("Number: ");
    long oneNumber = number;//oneNumber is for first operation
    long twoNumber = number;//twoNumber is for second operation

    long newNumber;
    int lastDigit, newLastDigit, sum, length;
    long firstTwo = number;
    long firstOne = number;
    int sum1=0;
    int sum2=0;

    //finding length
    length=log10(number) + 1;

    //first two digit and first one digit grapping
    while(firstTwo >= 100)
    {
        firstTwo = firstTwo/10;
    }
    firstOne = firstTwo/10;

    //sum of first operation-every other digit from the last digit
    do
    {
       lastDigit=oneNumber%10;
       newNumber=oneNumber/100;
       oneNumber=newNumber;

       sum1+=lastDigit;
    }
    while(newNumber>0);


    //sum of second operation-every other digit, starting with the number’s second-to-last digit
    twoNumber=number/10;
    do
    {
       lastDigit=twoNumber%10;
       newNumber=twoNumber/100;
       twoNumber=newNumber;
       newLastDigit=2*lastDigit;

       //spliting the number greater than ten
       if(newLastDigit>=10)
       {
           newLastDigit=newLastDigit%10+1;
       }
       sum2+=newLastDigit;
    }
    while(newNumber>0);

    //sum
    sum=sum1+sum2;

    if(sum%10==0)
    {

        if(length>=13 && length<=16)
        {
            if(firstTwo==34 || firstTwo==37)
            {
                printf("AMEX\n");
            }
            else if(firstTwo>=51 && firstTwo<=55)
            {
                printf("MASTERCARD\n");
            }
            else if(firstOne==4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }

        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }

}